makeStage2.sh

```diff
@@ -40,5 +41,100 @@ cd $usepop/pop/x/src/
 # $usepop/pop/src/newpop -link -nox -norsv
 # $usepop/pop/src/newpop -link -x=-xm -norsv
 # $usepop/pop/src/newpop -link -x=-xt -norsv
+
+mkdir -p "${BUILD_HOME}/environments"
+
+# We need to capture the Poplog environment for each build variant. This is
+# required for the poplog command tool, which needs to bind the appropriate 
+# environment variables. Each build variant has slightly different variables
+# and values. So we capture these and later on will synthesise them into 
+# C-code functions. This is the best point at which to capture the variables.
+
+# Note that we do not wish to capture the values of variables automatically
+# introduced by running a shell SHLVL and PWD. Nor do we want to capture
+# the folder location variables poplib, poplocalauto, poplocalbin. That is
+# because GetPoplog has different defaults for those and the old values are
+# irrelevant.
+
+# We run the popenv.sh script inside a clean environment to capture the 
+# set of environment variables needed. We then need to replace any matches
+# of the string "_build/poplog_base" ($usepop) with our unique value USEPOP.
+# This will allow us to dynamically substitute with the selfHome'd value
+# at run-time. 
+#
+# This is inherently a weak strategy because it relies on being able to 
+# identify substitutions of $usepop. We improve its robustness by doing the
+# process twice with different values of $usepop - using the '..' trick.
+# If the resultant code is not identical we have a problem and we halt.
+# (N.B It is probably not necessary to run the variables through sort but I 
+# couldn't find a clear guarantee that env generates a sorted list.)
+#
+echo_env() {
+    cmd='(usepop="'"$1"'" && . $usepop/pop/com/popenv.sh && env)'
+    env -i sh -c "$cmd" | sort | \
+    grep -v '^\(_\|SHLVL\|PWD\|poplib\|poplocal\(auto\|bin\)\?\)=' | \
+    sed -e 's!'"$1"'![//USEPOP//]!g' | \
+    sort
+}
+
+
+### nox ########################################################################
+
+# Newpop - see https://raw.githubusercontent.com/GetPoplog/Base/main/pop/help/newpop
+# Rebuilds $popsys: re-links basepop11, rebuild saved images and generate scripts.
+# -nox specifies basepop11 should not be linked against X-windows.
+$usepop/pop/src/newpop -link -nox -norsv
+
+echo_env "$usepop" > "${BUILD_HOME}/environments/nox-base"
+echo_env "$usepop/pop/.." > "${BUILD_HOME}/environments/nox-base-cmp"
+( cd "${BUILD_HOME}/environments" && \
+  sed -e 's!\[//USEPOP//]/pop/pop![//USEPOP//]/pop/pop-nox!g' < nox-base | \
+  sed -e 's!\[//USEPOP//]/pop/lib/psv/![//USEPOP//]/pop/lib/psv-nox/!g' > nox-new )
+
+mkdir -p "$usepop"/pop/pop-nox
+mkdir -p "$usepop"/pop/lib/psv-nox
+( cd "$usepop"/pop/pop; tar cf - . ) | ( cd "$usepop"/pop/pop-nox; tar xf - )
```

Comment:
Are we going to ditch this pattern for `cp -a`?

Reply:
There were problems adopting it because of issues with aborted builds. So I have abstracted out the pattern.